catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/DynamicSchemaResolver.java

```diff
@@ -226,33 +231,41 @@ public void addFields(Metacard metacard, SolrInputDocument solrInputDocument)
           AttributeFormat format = ad.getType().getAttributeFormat();
           String formatIndexName = ad.getName() + getFieldSuffix(format);
 
-          if (AttributeFormat.XML.equals(format)) {
+          if (AttributeFormat.XML.equals(format)
+              && solrInputDocument.getFieldValue(
+                      formatIndexName + getSpecialIndexSuffix(AttributeFormat.STRING))
+                  == null) {
             List<String> parsedTexts = parseTextFrom(attributeValues);
 
-            // text => metadata_txt_ws
-            String whitespaceTokenizedIndexName =
-                ad.getName()
-                    + getFieldSuffix(AttributeFormat.STRING)
-                    + SchemaFields.WHITESPACE_TEXT_SUFFIX;
-            solrInputDocument.addField(whitespaceTokenizedIndexName, parsedTexts);
-
-            // text => metadata_txt_ws_has_case
-            String whiteSpaceTokenizedHasCaseIndexName =
-                ad.getName()
-                    + getFieldSuffix(AttributeFormat.STRING)
-                    + SchemaFields.WHITESPACE_TEXT_SUFFIX
-                    + SchemaFields.HAS_CASE;
-            solrInputDocument.addField(whiteSpaceTokenizedHasCaseIndexName, parsedTexts);
-
-            // text => metadata_txt_tokenized
+            // parsedTexts => *_txt_tokenized
             String specialStringIndexName =
                 ad.getName()
                     + getFieldSuffix(AttributeFormat.STRING)
                     + getSpecialIndexSuffix(AttributeFormat.STRING);
             solrInputDocument.addField(specialStringIndexName, parsedTexts);
-
-            // text case sensitive
-            solrInputDocument.addField(specialStringIndexName + SchemaFields.HAS_CASE, parsedTexts);
+          } else if (AttributeFormat.STRING.equals(format)
+              && solrInputDocument.getFieldValue(
+                      ad.getName() + getFieldSuffix(AttributeFormat.STRING))
+                  == null) {
+            List<Serializable> truncatedValues =
+                attributeValues
+                    .stream()
+                    .map(
+                        value ->
+                            value != null
+                                ? truncateAsUTF8(value.toString(), TOKEN_MAXIMUM_BYTES)
+                                : value)
+                    .collect(Collectors.toList());
+            // *_txt
+            solrInputDocument.addField(
+                ad.getName() + getFieldSuffix(AttributeFormat.STRING), truncatedValues);
```

Comment:
❓ @pklinef Are we still also indexing this field in addition to the tokenized version below? Not if there's a reason to keep both indexed or if we could disable with our current field definitions anyways -- but hopefully this doesn't have a negative effect on the index size.

Reply:
We are indexing the tokenized and the un-tokenized version.  The un-tokenized version is indexed as a single token for equality searches.